stdlib/stdlib.ml

```diff
@@ -565,6 +565,9 @@ let exit retcode =
 
 let _ = register_named_value "Pervasives.do_at_exit" do_at_exit
 
+external major : unit -> unit = "caml_gc_major"
```

Comment:
This should be guarded by whether naked pointers checker is enabled. 

For this we need a way to check whether naked pointers checker has been enabled. See `caml_sys_const_ostype_unix` https://github.com/ocaml/ocaml/blob/trunk/runtime/sys.c#L624 for inspiration. Let's add a function `value caml_sys_const_naked_pointers_checked (value unit)` in `runtime/sys.c` that returns `true` if naked pointers are checked. Then you can change this code to be:

```ocaml
external major : unit -> unit = "caml_gc_major"
external naked_pointers_checked : unit -> bool = "caml_sys_const_naked_pointers_checked"
let _ = if naked_pointers_checked () then at_exit major
```

Unlike the `caml_sys_const_ostype_unix`, `caml_sys_const_naked_pointers_checked` need not be a primitive and can just be an external call. We're going to invoke it exactly once per run.